OrbitCore/LinuxEventTracer.cpp

```diff
@@ -0,0 +1,262 @@
+//-----------------------------------
+// Copyright Pierric Gimmig 2013-2019
+//-----------------------------------
+
+#include "ContextSwitch.h"
+#include "OrbitProcess.h"
+#include "PrintVar.h"
+#include "LinuxPerfUtils.h"
+#include "LinuxPerfEvent.h"
+#include "LinuxEventTracer.h"
+#include "LinuxUtils.h"
+#include "Utils.h"
+#include "Capture.h"
+#include "TimerManager.h"
+
+#include <assert.h>
+#include <functional>
+#include <queue>
+#include <vector>
+
+
+//-----------------------------------------------------------------------------
+LinuxEventTracer::LinuxEventTracer()
+{
+    //m_Callback = a_Callback ? a_Callback : [this](const std::string& a_Buffer)
+    //{
+    //    CommandCallback(a_Buffer);
+    //};
+}
+
+//-----------------------------------------------------------------------------
+void LinuxEventTracer::Start()
+{
+    PRINT_FUNC;
+#if __linux__
+    m_ExitRequested = false;
+
+    m_Thread = std::make_shared<std::thread>(&LinuxEventTracer::foo, &m_ExitRequested);
+
+   /* m_Thread = std::make_shared<std::thread>
+        ( &LinuxUtils::StreamCommandOutput
+        , "perf record -e sched:sched_switch -o - | perf script -i -"
+        , m_Callback
+        , &m_ExitRequested );
+    */
+    m_Thread->detach();
+#endif
+}
+
+//-----------------------------------------------------------------------------
+void LinuxEventTracer::Stop()
+{
+    m_ExitRequested = true;
+}
+
+
+//-----------------------------------------------------------------------------
+void LinuxEventTracer::CommandCallback(std::string a_Line)
+{
+    std::vector<std::string> tokens = Tokenize(a_Line);
+    
+    // TODO: log broken lines
+    if (tokens[4] != "sched:sched_switch:" || tokens[8] != "==>")
```

Comment:
Make sure you check the length of `tokens` before accessing it. O/w this line will cause a crash.